near-sdk/src/environment/env.rs

```diff
@@ -788,4 +825,40 @@ mod tests {
             .as_slice()
         );
     }
+
+    #[derive(Deserialize)]
+    struct EcrecoverTest {
+        #[serde(with = "hex::serde")]
+        m: [u8; 32],
+        v: u8,
+        #[serde(with = "hex::serde")]
+        sig: [u8; 64],
+        mc: bool,
+        #[serde(deserialize_with = "deserialize_option_hex")]
+        res: Option<[u8; 64]>,
+    }
+
+    fn deserialize_option_hex<'de, D, T>(deserializer: D) -> Result<Option<T>, D::Error>
+    where
+        D: Deserializer<'de>,
+        T: FromHex,
+        <T as FromHex>::Error: Display,
+    {
+        Deserialize::deserialize(deserializer)
+            .map(|v: Option<&str>| v.map(FromHex::from_hex).transpose().map_err(Error::custom))
+            .and_then(|v| v)
+    }
+
+    #[cfg(not(target_arch = "wasm32"))]
+    #[test]
+    fn test_ecrecover() {
+        for EcrecoverTest { m, v, sig, mc, res } in from_slice::<'_, Vec<_>>(
+            std::fs::read("tests/ecrecover-tests.json").unwrap().as_slice(),
+        )
+        .unwrap()
+        {
+            assert_eq!(super::ecrecover(&m, &sig, v, mc), res);
+            with_ext_cost_counter(|cc| cc.clear());
```

Comment:
I am having some troubles with making this test work as it runs out of gas during execution. I saw that `nearcore` tests use `with_ext_cost_counter(|cc| cc.clear())` to clear gas counter, but it does not seems to have an effect here...

Reply:
Indeed, thanks!